web-ui/src/main/resources/catalog/components/admin/harvester/HarvesterDirective.js

```diff
@@ -171,13 +171,31 @@
              };
              var defaultPrivileges = [getPrivilege(1)];
 
+             // deal with order by
+             scope.sorter = null
+             scope.setSorter = function(g) {
+               if (scope.sorter == 'name') return g.label ? g.label[scope.lang] : g.name;
+               else if (scope.sorter == 'checked') return scope.selectedPrivileges[g['@id']];
+               else return 0;
+             }
+
+             var internalGroups =  gnShareConstants.internalGroups;
+
+             scope.keepInternalGroups = function(g){
+               if (internalGroups.includes(parseInt(g['@id']))) return true;
+               else return false;
+             }
+             scope.removeInternalGroups = function(g){
+               if (internalGroups.includes(parseInt(g['@id']))) return false;
+               else return true;
+             }
              scope.visibleTo = function(who) {
                scope.custom = false;
                scope.selectedPrivileges = {};
-               if (who == 'all') {
+               if (this.all == 'all') {
                  scope.allGroup = false;
                  scope.selectedPrivileges = {1: true};
-               } else if (who == 'none') {
+               } else if (this.all === false && who === undefined) {
```

Comment:
I'm not really an expert in Javascript, so maybe it's ok, but this mix in `this.all` with string and boolean values it's a bit confusing, probably due to my Java background. 

Please, if looks ok fro Javascript, ignore this comment.